crates/aptos/src/move_tool/mod.rs

```diff
@@ -993,76 +972,277 @@ impl CliCommand<TransactionSummary> for UpgradeObjectPackage {
+    #[clap(long, value_parser = crate::common::types::load_account_arg)]
+    pub(crate) to: AccountAddress,
+    #[clap(flatten)]
+    pub(crate) override_size_check_option: OverrideSizeCheckOption,
+    #[clap(flatten)]
+    pub(crate) included_artifacts_args: IncludedArtifactsArgs,
+    #[clap(flatten)]
+    pub(crate) move_options: MovePackageDir,
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+}
+
+#[async_trait]
+impl CliCommand<TransactionSummary> for TransferCodeObject {
+    fn command_name(&self) -> &'static str {
+        "Transfer"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<TransactionSummary> {
+        let message = format!(
+            "Do you want to transfer ownership of the code package at object address {} to owner address {}?",
+            self.object_address,
+            self.to,
+        );
+        prompt_yes_with_override(&message, self.txn_options.prompt_options)?;
+
+        let payload =
+            aptos_cached_packages::aptos_stdlib::object_transfer_call(self.object_address, self.to);
+
+        submit_tx_and_check(
+            &self.txn_options,
+            payload,
+            &self.object_address.to_string(),
+            self.override_size_check_option.override_size_check,
+            "Code ownership was successfully transferred to {}.",
+        )
+        .await
+    }
+}
+
+fn build_package_options(
+    move_options: &MovePackageDir,
+    included_artifacts_args: &IncludedArtifactsArgs,
+) -> anyhow::Result<BuiltPackage> {
+    let options = included_artifacts_args.included_artifacts.build_options(
+        move_options.dev,
+        move_options.skip_fetch_latest_git_deps,
+        move_options.named_addresses(),
+        move_options.bytecode_version,
+        move_options.compiler_version,
+        move_options.language_version,
+        move_options.skip_attribute_checks,
+        move_options.check_test_code,
+    );
+    BuiltPackage::build(move_options.get_package_path()?, options)
+}
+
+async fn submit_tx_and_check(
+    txn_options: &TransactionOptions,
+    payload: TransactionPayload,
+    object_address: &str,
+    override_size_check: bool,
+    success_message: &str,
+) -> CliTypedResult<TransactionSummary> {
+    let size = bcs::serialized_size(&payload)?;
+    println!("package size {} bytes", size);
+
+    if !override_size_check && size > MAX_PUBLISH_PACKAGE_SIZE {
+        return Err(CliError::UnexpectedError(format!(
+            "The package is larger than {} bytes ({} bytes)! To lower the size \
+            you may want to include fewer artifacts via `--included-artifacts`. \
+            You can also override this check with `--override-size-check",
+            MAX_PUBLISH_PACKAGE_SIZE, size
+        )));
+    }
+
+    let result = txn_options
+        .submit_transaction(payload)
+        .await
+        .map(TransactionSummary::from);
+
+    if result.is_ok() {
+        println!("{} {}", success_message, object_address);
```

Comment:
feels kind of like a weird pattern. We should probably have success and failure fns on the trait and dispatch appropriately on failure.

now we have a println here randomly...